src/test/TestUtils.cpp

```diff
@@ -100,18 +100,15 @@ TestApplication::createInvariantManager()
 }
 
 time_t
-getTestDate(int day, int month, int year)
+getTestDateTime(int day, int month, int year, int hour, int minute, int second)
```

Comment:
Yeah, you're right, I ended up not needing those.  I've removed nearly all of the time-related changes.

However, I've kept one, and to make sure that it's tested and to illustrate why I like it, I've also added uses of it in a couple of tests.  (It's factored out into a separate commit now, so that it's straightforward to review in isolation and to remove if you still don't like it.)   It's the change to add an overload of `closeLedgerOn()` to take a `time_t`.

The accompanying test case changes are to code that confused me when I first saw it, back when I knew practically nothing about the protocol.  I was trying to figure out what the significance was of the number of days between ledger closes -- whether there was some relationship, for example, with the number of seconds from the start time to the transaction's `minTime` or `maxTime`.

It turns out that there _is_ no significance to the number of days!  Any time later than the first ledger close time would have been fine.  The old `closeLedgerOn()` interface just doesn't give you any way of saying "one moment later than <some time that we already have stored in a variable>".  So using the new overload allows removing some hard-coded constants that actually cost me some comprehension time.

The last use of the new overload in the diff shows another way in which it can shorten existing test code if that code needs a `time_t` representation of the last ledger close time.  The old code did the day/month/year -> `time_t` conversion awkwardly: by calling `closeLedgerOn()` with a day/month/year, and then getting the `time_t` from `app->getLedgerManager().getLastClosedLedgerHeader().header.scpValue.closeTime`.  So it used `closeLedgerOn()` to do the conversion implicitly, and read the output from one of `closeLedgerOn()`'s side effects.  The new overload allows us to do the conversion explicitly, with `getTestDate()`, and then use the result of that both as a parameter to `closeLedgerOn()` and later in the test code where it needs a `time_t`.

I feel that both of those changes make the test code shorter and clearer, and justify the new overload of `closeLedgerOn()`.